common/persistence/telemetry/cluster_metadata_store_gen.go

```diff
@@ -0,0 +1,143 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Code generated by gowrap. DO NOT EDIT.
+// template: gowrap_template
+// gowrap: http://github.com/hexdigest/gowrap
+
+package telemetry
+
+//go:generate gowrap gen -p go.temporal.io/server/common/persistence -i ClusterMetadataStore -t gowrap_template -o cluster_metadata_store_gen.go -l ""
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/trace"
+	_sourcePersistence "go.temporal.io/server/common/persistence"
+)
+
+// telemetryClusterMetadataStore implements ClusterMetadataStore interface instrumented with OpenTelemetry.
+type telemetryClusterMetadataStore struct {
+	_sourcePersistence.ClusterMetadataStore
+	tracer trace.Tracer
+}
+
+// newTelemetryClusterMetadataStore returns telemetryClusterMetadataStore.
+func newTelemetryClusterMetadataStore(base _sourcePersistence.ClusterMetadataStore, tracer trace.Tracer) telemetryClusterMetadataStore {
+	return telemetryClusterMetadataStore{
+		ClusterMetadataStore: base,
+		tracer:               tracer,
+	}
+}
+
+// DeleteClusterMetadata wraps ClusterMetadataStore.DeleteClusterMetadata.
+func (_d telemetryClusterMetadataStore) DeleteClusterMetadata(ctx context.Context, request *_sourcePersistence.InternalDeleteClusterMetadataRequest) (err error) {
```

Comment:
`_` is against go convention, but ok, I will leave it for code generators.

Reply:
Good point. This comes from their templates. I'm happy to change it to `d`.